File: CPP/BubbleSort.cpp
#include <iostream>
#include <algorithm>
using namespace std;

//O(n^2)
void bubbleSort(int tab[], int n) {
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n - i - 1; j++)
            if (tab[j] > tab[j + 1])
                swap(tab[j], tab[j + 1]);
}

int main() {
    int tab[] = {3, 8, 17, 99, 2, 10, 77};
    cout << "Before sort: ";
    for (int i = 0; i < 7; i++)
        cout << tab[i] << ", ";
    cout << endl;
    bubbleSort(tab, 7);
    cout << "After sort: ";
    for (int i = 0; i < 7; i++)
        cout << tab[i] << ", ";
}

File: Python/BubbleSort.py
def bubble_sort(tab: list) -> None:
    for i in range(len(tab)):
        for j in range(len(tab) - i - 1):
            if tab[j] > tab[j + 1]:
                tab[j], tab[j + 1] = tab[j + 1], tab[j]

tab = [3, 10, 77, 99, 2, 8, 17]
print("Before sort:", tab)
bubble_sort(tab)
print("After sort:", tab)